client/competition.js

```diff
@@ -7,7 +7,14 @@ Template.competition.helpers({
       fields: { _id: 1 }
     }).fetch();
 
-    var registered = _.contains(_.pluck(registrations, '_id'), Meteor.userId());
+    var registered = _.contains(_.pluck(registrations, '_id'), Meteor.userId()); // TODO Does this even work?
```

Comment:
I haven't run the code, so this is just from reading it. Lazy but true.

The first query seems like all you need. If it returns anything, the user is registered for the competition, otherwise not.

But then we check that the **Registrations _id** is equal to **Meteor.userId()**, and that just seems non sensical. And always false.